### PR TITLE
apis: exclude slo pkg imports from extension to avoid cycle ref

### PR DESCRIPTION
**Pull request by zwzhang0107**
### Ⅰ. Describe what this PR does

exclude the imports of `slo` pkg from `extension` pkg

### Ⅱ. Does this pull request fix one issue?

extension pkg is a basic pkg, it should not import other pkg like slo. Otherwise it will cause cycle reference.

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: apis/slo/v1alpha1/pod.go [hunk 1]
```diff
@@ -14,33 +14,32 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package extension
+package v1alpha1
```
Comment (eahydra) [inline apis/slo/v1alpha1/pod.go:17]: I have a little doubt: will some codes in the extension package be adjusted according to this new layout? For example, slo-related extension protocols are putting into slo/v1alpha1, and scheduling-related extension protocols are putting into scheduling/v1alpha1.

File: apis/slo/v1alpha1/pod.go [hunk 2]
```diff
@@ -14,33 +14,32 @@ See the License for the specific language governing permissions and
...
 
 import (
 	"encoding/json"
+	apiext "github.com/koordinator-sh/koordinator/apis/extension"
```
Comment (saintube) [inline apis/slo/v1alpha1/pod.go:21]: need goimports

File: apis/slo/v1alpha1/pod.go (hunk 1 above)
Comment (zwzhang0107) [reply apis/slo/v1alpha1/pod.go:17]: i think it depends, for example the `CPUBurstConfig` struct, it is slo-relate and specific for slo config, which already defined in slo/v1alpha1 and no one else will use it except slo-controller/koordlet. It's better to mv related functions to slo/v1alpha1. While for other common slo-related functions like `GetPodQoSClassByName`, which can be used for other modules, maybe should be kept inside extension.

Comment (eahydra) [reply apis/slo/v1alpha1/pod.go:17]: If it is determined to adjust the code layout as a whole, I think it is OK.

**Review by eahydra — APPROVED** [review]
/lgtm

**Review by saintube — APPROVED** [review]
/lgtm

**Review by jasonliu747 — COMMENTED** [review]
/lgtm

**Review by hormes — APPROVED** [review]

**Pull request merged**